packages/contracts-bedrock/src/governance/ProposalValidator.sol

```diff
@@ -0,0 +1,1155 @@
+            revert ProposalValidator_ProposalAlreadySubmitted();
+        }
+
+        // Store proposal metadata
+        proposal.proposer = _msgSender();
+        proposal.proposalType = ProposalType.CouncilMemberElections;
+        proposal.votingCycle = _votingCycle;
+
+        emit ProposalSubmitted(proposalId_, _msgSender(), _proposalDescription, ProposalType.CouncilMemberElections);
+        emit ProposalVotingModuleData(proposalId_, proposalVotingModuleData);
+    }
+
+    /// @notice Submits a GovernanceFund or CouncilBudget proposal type that transfers OP tokens for approval and
+    /// voting.
+    /// @dev For UI integration: Frontend interfaces should present this as a percentage input to users (e.g., "25%"),
+    /// then convert to the absolute vote count by calculating: (percentage / 100) * total_votable_supply.
+    /// Direct contract callers must provide the absolute number of votes required for passage.
+    /// @param _criteriaValue The absolute number of votes required for the proposal to pass. This represents the
+    /// threshold that must be met or exceeded for any option to be considered successful.
+    /// @param _optionsDescriptions The strings of the different options that can be voted.
+    /// @param _optionsRecipients An address for each option to transfer funds to in case the option passes the voting.
+    /// @param _optionsAmounts The amount to transfer for each option in case the option passes the voting.
+    /// @param _description Description of the proposal.
+    /// @param _proposalType The type of proposal (must be GovernanceFund or CouncilBudget).
+    /// @param _votingCycle The voting cycle number the proposal is targetted for.
+    /// @return proposalId_ The ID of the submitted proposal.
+    function submitFundingProposal(
+        uint128 _criteriaValue,
+        string[] memory _optionsDescriptions,
+        address[] memory _optionsRecipients,
+        uint256[] memory _optionsAmounts,
+        string memory _description,
+        ProposalType _proposalType,
+        uint256 _votingCycle
+    )
+        external
+        returns (uint256 proposalId_)
+    {
+        // Only funding proposal types can use this function
+        if (_proposalType != ProposalType.GovernanceFund && _proposalType != ProposalType.CouncilBudget) {
+            revert ProposalValidator_InvalidFundingProposalType();
+        }
+
+        // Validate voting cycle exists and is not in the past
+        VotingCycleData memory votingCycleData = votingCycles[_votingCycle];
+        if (votingCycleData.startingTimestamp == 0 || votingCycleData.startingTimestamp < block.timestamp) {
+            revert ProposalValidator_InvalidVotingCycle();
+        }
+
+        // Validate input arrays have matching lengths
+        uint256 optionsLength = _optionsDescriptions.length;
+        if (optionsLength != _optionsRecipients.length || optionsLength != _optionsAmounts.length) {
```

Comment:
This might be a small nit but instead of imposing this check in submitting the proposal should this be moved to `_buildApprovalModuleOptions` just because this is an always required to check in the approval module and then if we add different proposal types it reduces a duplicated code pattern. Same thing for the `optionsLength`.

Reply:
Hmmm I am not sure about this check because this is not needed for example on the Council Member elections even though it uses the Approval module, but I believe we should add the check bellow now that you mentioned it `if (optionsLength == 0 || optionsLength > type(uint8).max)`